### PR TITLE
docs: update docs for bigquery sink json support

### PR DESCRIPTION
**Pull request by kevinbheda**

### CODE REVIEW

File: docs/docs/sinks/bigquery-sink.md [hunk 1]
```diff
@@ -1,6 +1,45 @@
...
+
+## At Least Once Guarantee
+
+Because of asynchronous consumer mode and the possibility of retry on the insert operation. There is no guarantee of the message order that successfully sent to the sink.
+That also happened with commit offset, the there is no order of the offset number of the processed messages.
+Firehose collect all the offset sort them and only commit the latest continuous offset.
+This will ensure all the offset being committed after messages successfully processed even when some messages are being re processed by retry handler or when the insert operation took a long time.
+
+## Configurations
+A Bigquery sink Firehose \(`SINK_TYPE`=`bigquery`\) Along with generic ones mentioned in [generic.md](../advance/generic.md), please refer to addtional configs mentioned in depot [Depot-configuration/bigquery-sink.md section](https://github.com/odpf/depot/blob/main/docs/reference/configuration/bigquery-sink.md)
```
Comment (prakharmathur82) [inline docs/docs/sinks/bigquery-sink.md:21]: this line can be better phrased

File: docs/docs/sinks/bigquery-sink.md [hunk 2]
```diff
@@ -1,6 +1,45 @@
...
+
+
```
Comment (prakharmathur82) [inline docs/docs/sinks/bigquery-sink.md:23]: we can talk about default columns in json as well

**Review by prakharmathur82 — CHANGES_REQUESTED** [review]

File: docs/docs/introduction.md [hunk 3]
```diff
@@ -19,6 +19,10 @@ Discover why users choose Firehose as their main Kafka Consumer
 - **Runtime** Firehose can run inside containers or VMs in a fully managed runtime environment like Kubernetes.
 - **Metrics** Always know what’s going on with your deployment with built-in monitoring of throughput, response times, errors, and more.
 
+## Supported Incoming data types from kafka
+- [Protobuf](https://developers.google.com/protocol-buffers)
+- [JSON](https://www.json.org/json-en.html)
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply docs/docs/introduction.md:24]: yes

File: docs/docs/sinks/bigquery-sink.md [hunk 4]
```diff
@@ -1,6 +1,45 @@
 # BigQuery
 
-A Bigquery sink Firehose \(`SINK_TYPE`=`bigquery`\) requires env variables to be set along with Generic ones and 
-env variables in depot repository. The Firehose sink uses bigquery implementation available [depot](https://github.com/odpf/depot) repository. 
+Bigquery Sink has several responsibilities, first creation of bigquery table and dataset when they are not exist, second update the bigquery table schema based on the latest protobuf schema, third translate protobuf messages into bigquery records and insert them to bigquery tables.
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply docs/docs/sinks/bigquery-sink.md:3]: updated the sentence

File: docs/docs/sinks/bigquery-sink.md (hunk 1 above)
Comment (kevinbheda) [reply docs/docs/sinks/bigquery-sink.md:21]: tried re-phrasing it, let me know if it looks good

File: docs/docs/sinks/bigquery-sink.md (hunk 2 above)
Comment (kevinbheda) [reply docs/docs/sinks/bigquery-sink.md:23]: added the section

File: docs/docs/concepts/architecture.md [hunk 5]
```diff
@@ -71,14 +71,18 @@ The final state of message can be any one of the followings after it is consumed
...
-- Protocol buffers are Google's language-neutral, platform-neutral, extensible mechanism for serializing structured data. Data streams on Kafka topics are bound to a protobuf schema.
-- Firehose deserializes the data consumed from the topics using the Protobuf descriptors generated out of the artifacts. The artifactory is an HTTP interface that Firehose uses to deserialize.
-- The schema handling ie., find the mapped schema for the topic, downloading the descriptors, and dynamically being notified of/updating with the latest schema is abstracted through the Stencil library.
+    The Stencil is a proprietary library that provides an abstraction layer, for schema handling.
 
-  The Stencil is a proprietary library that provides an abstraction layer, for schema handling.
+    Schema Caching, dynamic schema updates, etc. are features of the stencil client library.
 
-  Schema Caching, dynamic schema updates, etc. are features of the stencil client library.
+- Incase when `INPUT_SCHEMA_DATA_TYPE is set to json`
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply docs/docs/concepts/architecture.md:83]: updated the description

**Review by prakharmathur82 — APPROVED** [review]

**Pull request merged**